packages/core/src/schema/models/documentRunErrors.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  bigint,
+  bigserial,
+  index,
+  jsonb,
+  text,
+  uuid,
+} from 'drizzle-orm/pg-core'
+
+import { DocumentRunErrorCodes } from '../../constants'
+import { latitudeSchema } from '../db-schema'
+import { timestamps } from '../schemaHelpers'
+import { commits } from './commits'
+
+export const documentRunErrorCodeEnum = latitudeSchema.enum('document_run_error_code_enum', [
+  DocumentRunErrorCodes.Unknown,
+  DocumentRunErrorCodes.DefaultProviderExceededQuota,
+  DocumentRunErrorCodes.DocumentConfigError,
+  DocumentRunErrorCodes.MissingProvider,
+  DocumentRunErrorCodes.ChainCompileError,
+  DocumentRunErrorCodes.AIRunError,
+])
+
+export const documentRunErrors = latitudeSchema.table(
+  'document_run_errors',
+  {
+    id: bigserial('id', { mode: 'number' }).notNull().primaryKey(),
+    code: documentRunErrorCodeEnum('code').notNull(),
+    documentUuid: uuid('document_uuid').notNull(),
+    commitId: bigint('commit_id', { mode: 'number' })
+      .notNull()
+      .references(() => commits.id, {
+        onDelete: 'restrict',
+        onUpdate: 'cascade',
+      }),
+    message: text('message').notNull(),
+    details: jsonb('details').$type<Record<string, unknown>>(),
+    ...timestamps(),
+  },
+  (table) => ({
+    documentLogUuidIdx: index('document_run_errors_uuid_idx').on(
+      table.documentUuid,
+    ),
+    commitIdIdx: index('document_run_errors_commit_id_idx').on(table.commitId),
+  }),
+)
```

Comment:
What if the error comes from an evaluation? That's not a document run but still has the same behaviour.
I think this should have a polymorphic relation with either `documentLog` or `evaluationResult`, since they're the two sources of provider logs and AI execution.

Reply:
This is similar as how `ProviderLogs` should have a polymorphic relationship with either `DocumentLog` or `EvaluationResult` too, instead of our current approach.

Reply:
I think that makes a lot of sense. Changing